### PR TITLE
Add minikube from source

### PR DESCRIPTION
**Pull request by xmnlab**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [ ] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by xmnlab** [comment]
@esloch I am adding you as a maintainer of this package. please confirm if you are OK about that.

**Comment by esloch** [comment]
OK. Thanks @xmnlab.

**Comment by xmnlab** [comment]
it is done for review!

**Comment by xmnlab** [comment]
@conda-forge/staged-recipes it is done for review.

**Comment by xmnlab** [comment]
@ocefpaf at the end I could package that from the source :)

**Comment by sodre** [comment]
@xmnlab, 

Can you take a look at this [other `go` based recipe](https://github.com/conda-forge/consul-feedstock/blob/master/recipe/meta.yaml) for some ideas on how to reduce the amount of code in the recipe ?

At minimum we should be using the compiler tag syntax.

**Comment by xmnlab** [comment]
@sodre thanks for the feedback.

actually I am not a GO or minikube expert ... and I don't have much time to play with that ... 

basically the code used here was just a `make` and `mv`

that is the instruction specified on the documentation https://minikube.sigs.k8s.io/docs/contributing/building/

I didn't see different instructions to build just using `go install`

if you could help I really appreciate that.

another option would be re-bundle the binary to start with a minikube package and when we have more time work to have a better package from source.

File: recipes/minikube/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    - patch/go.sum.patch
+    - patch/go.mod.patch
+
+build:
+  number: 0
+  skip: true  # [not linux]
+
+requirements:
+  build:
+    - go
```
Comment (chrisburr) [inline recipes/minikube/meta.yaml]: I think this should use the compiler syntax:
```suggestion
    - {{ compiler("go") }}
```

File: recipes/minikube/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  run:
+    - kubernetes
+
+test:
+  commands:
+    - kubectl version --client
```
Comment (chrisburr) [inline recipes/minikube/meta.yaml]: This looks like it isn't testing this recipe
```suggestion
```

File: recipes/minikube/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
```
Comment (chrisburr) [inline recipes/minikube/meta.yaml]: Can you delete these comments from the example

File: recipes/minikube/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    - patch/go.mod.patch
+
+build:
+  number: 0
+  skip: true  # [not linux]
+
+requirements:
+  build:
+    - go
+  run:
```
Comment (chrisburr) [inline recipes/minikube/meta.yaml]: While you can get make from the host container it's better to get it from conda-forge:
```suggestion
    - make
  run:
```

File: recipes/minikube/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+source:
+  url: https://github.com/kubernetes/{{ name }}/archive/{{ version }}.tar.gz
+  sha256: 7aa57e5896852c499f1687fbc424abf93645e1801fc9f8c2833e0affbb76eb41
+  patches:
+    - patch/go.sum.patch
+    - patch/go.mod.patch
+
+build:
+  number: 0
+  skip: true  # [not linux]
```
Comment (chrisburr) [inline recipes/minikube/meta.yaml]: Have you tried building for other platforms? It's okay if it's too much work but I wouldn't be surprised if macOS just works.

File: recipes/minikube/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,51 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
+
+# Jinja variables help maintain the recipe as you'll update the version only here.
+# Using the name variable with the URL in line 14 is convenient
+# when copying and pasting from another recipe, but not really needed.
+{% set name = "minikube" %}
+{% set version = "v1.3.1" %}
```
Comment (chrisburr) [inline recipes/minikube/meta.yaml]: Remove `v` from the version number and add it to the URL:
```suggestion
{% set version = "1.3.1" %}
```

File: recipes/minikube/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/kubernetes/{{ name }}/archive/{{ version }}.tar.gz
```
Comment (chrisburr) [inline recipes/minikube/meta.yaml]:
```suggestion
  url: https://github.com/kubernetes/{{ name }}/archive/v{{ version }}.tar.gz
```

File: recipes/minikube/meta.yaml (hunk 3 above)
Comment (xmnlab) [reply recipes/minikube/meta.yaml]: sure .. thanks! I will do it!

File: recipes/minikube/meta.yaml (hunk 6 above)
Comment (xmnlab) [reply recipes/minikube/meta.yaml]: sounds good. thanks.

File: recipes/minikube/meta.yaml (hunk 7 above)
Comment (xmnlab) [reply recipes/minikube/meta.yaml]: I will do it. thanks!

File: recipes/minikube/meta.yaml (hunk 5 above)
Comment (xmnlab) [reply recipes/minikube/meta.yaml]: sounds good I  can try it.

File: recipes/minikube/meta.yaml (hunk 1 above)
Comment (xmnlab) [reply recipes/minikube/meta.yaml]: oh I understood now what @sodre said about this ... thanks for the explanation!

File: recipes/minikube/meta.yaml (hunk 4 above)
Comment (xmnlab) [reply recipes/minikube/meta.yaml]: it makes sense. I will add it. thanks!

File: recipes/minikube/meta.yaml (hunk 2 above)
Comment (xmnlab) [reply recipes/minikube/meta.yaml]: sounds good. I will remove it. thanks

**Comment by ocefpaf** [comment]
> @ocefpaf at the end I could package that from the source :)

I never doubt it :wink:

File: recipes/minikube/build.sh [hunk 8]
```diff
@@ -0,0 +1,12 @@
+#!/bin/env bash
+
+set -ex
+
+export GOPATH="${BUILD_PREFIX}/bin"
+export GOROOT="${BUILD_PREFIX}/go"
```
Comment (ocefpaf) [inline recipes/minikube/build.sh]: I wonder if those should be set automatically for conda-forge.

Comment (sodre) [reply recipes/minikube/build.sh]: These are already set from a conda-forge. This is why I referred them to the consul-feedstock as an example.

Comment (xmnlab) [reply recipes/minikube/build.sh]: thanks for the feedback @ocefpaf and @sodre ... that makes sense we can address that on the next PR in the feedstock repo.

**Pull request merged**